packages/react-instantsearch/src/core/InstantSearch.js

```diff
@@ -137,12 +138,22 @@ class InstantSearch extends Component {
   }
 
   getChildContext() {
+    // If not already cached, cache the bound methods so that we can forward them as part
+    // of the context.
+    if (!this._aisContextCache) {
+      this._aisContextCache = {
+        ais: {
+          onInternalStateUpdate: this.onWidgetsInternalStateUpdate.bind(this),
+          createHrefForState: this.createHrefForState.bind(this),
```

Comment:
Usually you'd just bind/reassign them in the constructor.

```js
constructor(props) {
  /* ... */
  this.onWidgetsInternalStateUpdate = this.onWidgetsInternalStateUpdate.bind(this);
  // etc.
}
```

Is there a particular reason we want to avoid the `method = () =>` construct?